File: gyroTest/src/main/java/com/stulnikov/arduino/arduinoBt/gyrotest/gyrotest/dagger/ApplicationModule.java
package com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.dagger;

import android.app.Application;

import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.bluetooth.BluetoothController;
import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.bluetooth.BluetoothImpl;
import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.sensor.AccelerometerProvider;
import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.sensor.SensorController;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author alexeystulnikov 2/28/17.
 */
@Module
public class ApplicationModule {

    private Application mApplication;

    public ApplicationModule(Application application) {
        mApplication = application;
    }

    @Provides
    @Singleton
    Application providesApplication() {
        return mApplication;
    }

    @Provides
    AccelerometerProvider provideAccelerometer(Application application) {
        return new SensorController(application);
    }

    @Provides
    BluetoothController privideBluetooth(){
        return new BluetoothImpl();
    }
}

File: gyroTest/src/main/kotlin/com/stulnikov/arduino/arduinoBt/gyrotest/gyrotest/dagger/ApplicationModule.kt
package com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.dagger

import android.app.Application
import android.content.Context
import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.bluetooth.BluetoothController
import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.bluetooth.BluetoothImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author alexeystulnikov 2/28/17.
 */
@Module
class ApplicationModule(private val application: Application) {

    @Provides
    @Singleton
    fun provideApplication(): Application {
        return application
    }

    @Provides
    @Singleton
    fun provideApplicationContext(): Context {
        return application.applicationContext
    }

    @Provides
    fun provideBluetooth(): BluetoothController {
        return BluetoothImpl()
    }
}